src/PIL/Image.py

```diff
@@ -502,10 +510,36 @@ def height(self):
     def size(self):
         return self._size
 
+    @property
+    def _size(self):
+        if self._use_im_values():
+            return self.im.size
+        return self.__size
```

Comment:
Why this can't be just in `size` property?

Reply:
The `_size` getter is actually used in quite a few places, so that would have to be changed to use the `size` getter first. It would save three lines if the `_size` getter was removed though:
```python
    @property
    def size(self):
        if self._use_im_values():
            return self.im.size
        return self.__size

    def _size(self, value):
        # set im.size first in case it raises an exception
        if self._use_im_values():
            self.im.size = value
        self.__size = value

    _size = property(fset=_size)
```